frontend/src/components/common/header/components/menu-popup/menu-popup.tsx

```diff
@@ -7,12 +7,17 @@ import styles from './styles.module.scss';
 
 type Props = {
   onClose: () => void;
+  initiator: React.RefObject<HTMLButtonElement>;
 };
 
-const Popup: FC<Props> = ({ onClose }) => {
+const Popup: FC<Props> = ({ onClose, initiator }) => {
   const dispatch = useAppDispatch();
   const popupRef = useRef<HTMLDivElement>(null);
-  useHandleClickOutside(popupRef, onClose);
+  useHandleClickOutside({
+    ref: popupRef,
+    onClick: onClose,
+    ignoredElement: initiator,
```

Comment:
```suggestion
```
now you don't need this functionality at all
if any, you can just disable the popup button while the popup is open

Reply:
The issue is that, `mousedown` event (of click outside) is fired before the `click` event (of button), when user clicks on button, when the popup is open, it first closes (popup is not visible, button is not disabled), then reopens (click event of button is fired)

Reply:
Also, when using `click` event, instead of `mousedown`, both events are fired, when trying to open the popup and it never opens

Reply:
its okey for now

Reply:
remove `ignoredElement`

Reply:
Done